x-pack/test/apm_api_integration/tests/errors/group_id.ts

```diff
@@ -0,0 +1,142 @@
+    overrides?: RecursivePartial<
+      APIClientRequestParamsOf<'GET /internal/apm/services/{serviceName}/errors/{groupId}'>['params']
+    >
+  ) {
+    const response = await apmApiClient.readUser({
+      endpoint: 'GET /internal/apm/services/{serviceName}/errors/{groupId}',
+      params: {
+        path: {
+          serviceName,
+          groupId: 'foo',
+          ...overrides?.path,
+        },
+        query: {
+          start: new Date(start).toISOString(),
+          end: new Date(end).toISOString(),
+          environment: 'ENVIRONMENT_ALL',
+          kuery: '',
+          ...overrides?.query,
+        },
+      },
+    });
+    return response;
+  }
+
+  registry.when('when data is not loaded', { config: 'basic', archives: [] }, () => {
+    it('handles the empty state', async () => {
+      const response = await callApi();
+      expect(response.status).to.be(200);
+      expect(response.body.occurrencesCount).to.be(0);
+    });
+  });
+
+  registry.when(
+    'when data is loaded',
+    { config: 'basic', archives: ['apm_mappings_only_8.0.0'] },
+    () => {
+      describe('error group id', () => {
+        const appleTransaction = {
+          name: 'GET /apple 🍎 ',
+          successRate: 75,
+          failureRate: 25,
+        };
+        const bananaTransaction = {
+          name: 'GET /banana 🍌',
+          successRate: 50,
+          failureRate: 50,
+        };
+
+        before(async () => {
+          const serviceGoProdInstance = service(serviceName, 'production', 'go').instance(
+            'instance-a'
+          );
+
+          const interval = '1m';
+
+          const documents = [appleTransaction, bananaTransaction]
+            .map((transaction, index) => {
+              return [
+                ...timerange(start, end)
+                  .interval(interval)
+                  .rate(transaction.successRate)
+                  .flatMap((timestamp) =>
+                    serviceGoProdInstance
+                      .transaction(transaction.name)
+                      .timestamp(timestamp)
+                      .duration(1000)
+                      .success()
+                      .serialize()
+                  ),
+                ...timerange(start, end)
+                  .interval(interval)
+                  .rate(transaction.failureRate)
+                  .flatMap((timestamp) =>
+                    serviceGoProdInstance
+                      .transaction(transaction.name)
+                      .errors(
+                        serviceGoProdInstance
+                          .error(`Error ${index}`, transaction.name)
+                          .timestamp(timestamp)
+                      )
+                      .duration(1000)
+                      .timestamp(timestamp)
+                      .failure()
+                      .serialize()
+                  ),
+              ];
+            })
+            .flatMap((_) => _);
+
+          await synthtraceEsClient.index(documents);
```

Comment:
I noticed how both you and @gbamparop created a separate file `generate_data` for each endpoint. I like this approach because it makes it easier to re-use the specific data generator, as well as making tests cleaner since they don't contain datagen code.

Example: https://github.com/elastic/kibana/tree/2cfcb8caf938c3630474f0fc7c161597168dbd67/x-pack/test/apm_api_integration/tests/services/error_groups